### ISSUE TITLE
Crashing issue in -[Appboy userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:]

### ISSUE THREAD

**Issue by chuganzy** [Appboy/appboy-ios-sdk#223]
* [x] I have checked [Braze's documentation](https://www.braze.com/documentation/iOS/) and am still experiencing the issue.

# Report

As titled.
Please refer to the crash log below.

## Describe your environment.

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | iOS 11 - 13
 SDK Version             | 3.20.2
 Integration Method      | CocoaPods
 Xcode Version           | 11.3.1
 Repro rate              | One of our top crashes
 
## What did you do?

Crash issues are reported.

## What did you expect to happen?

No crash.

## What happened instead?

Crashing.

## Steps to reproduce

n/s

## Code Snippet

n/a

## Are you doing any feature customizations that may relate to the issue? Can you share the code snippet?

no

## Project that demonstrates the issue

https://apps.apple.com/us/app/mercari-the-selling-app/id896130944

## Crash log

We use `UserNotifications` framework to receive push notifications.

```
Appboy.m - Line 972
-[Appboy userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler:] + 972

Fatal Exception: NSInvalidArgumentException
-[NSNull isEqualToString:]: unrecognized selector sent to instance XXXXX
```

**Comment by hokstuff** [comment]
Hi @chuganzy,

Thanks for filing this. We are working on some updates that we aim to release in our next SDK version that should alleviate the crashes. In the meantime, we are releasing a backend patch that should be live tomorrow which we expect should stop the crashes as well.

Best,
Daniel

**Comment by hokstuff** [comment]
We have now released [version 3.22.0 of our SDK](https://github.com/Appboy/appboy-ios-sdk/releases/tag/3.22.0) that includes an SDK-side patch which will alleviate the crashes. Let us know if you still run into any more issues!
